### ISSUE TITLE
Bug. Virtual populate for Discriminator in Mongoose 4.10.*

### ISSUE THREAD

**Issue by chichivica** [Automattic/mongoose#5314]
Hi guys, thanks a lot for the greatest ORM for nodejs ever. 
We faced some bug which easy to reproduce:

  ```
let AuthorSchema = mongoose.Schema({
    name: String,
    code: Number
  });
  let Author = mongoose.model('Author', AuthorSchema);
  let PostSchema = mongoose.Schema({
    text: String,
    authorCode: Number
  });
  PostSchema.virtual('refAuthor', {
    ref: 'Author',
    localField: 'authorCode',
    foreignField: 'code',
    justOne: true
  });

  let SubPostSchema = mongoose.Schema({
    about: String
  });

  let Post = mongoose.model('Post', PostSchema);
  let SubPost = Post.discriminator('SubPost', SubPostSchema);

  // Author.create({
  //   name: 'Ivan',
  //   code: '123'
  // })
  //   .then(author => {
  //     return SubPost.create({
  //       text: 'lorem ipsum dolor',
  //       authorCode: 123,
  //       about: 'subpost about the post'
  //     })
  //   })
  SubPost.findOne()
    .then(subPost => {
      Post.populate(subPost, 'refAuthor')
        .then(_popuated => {
          assert.equal(_popuated.refAuthor.name, 'Ivan'); //returns undefined
          assert.equal(subPost.refAuthor.name, 'Ivan'); //returns undefined
        })
    });
  SubPost.findOne()
    .populate('refAuthor')
    .then(subPost => {
      assert.equal(subPost.refAuthor.name, 'Ivan'); //works great! 
    });

 ```
That behavior was found only in  4.10.X, in **4.9.X everything works fine.** 
At the beginning I thought that the problem consists MongoDb and tested that with 3.2.1 3.3.2 3.4.4 -- no results.
At the closer look I found that in the `_popuated` and `subPost` (first findOne) `$$populatedVirtuals` property set as necessary, but its content doesn't appear in the document itself. 

Actually I didn't spend much time to investigate the problem but it looks like that `assignVals` function in model.js:3145 set populated values properly but because that virtual created in the parent model `Post` and does not belongs to the heir `SubPost` they are disposed.

Bests,
Ivan

**Comment by SBRK** [comment]
This should be resolved in next release https://github.com/Automattic/mongoose/issues/5311

**Comment by vkarpov15** [comment]
Yeah this looks like a duplicate of #5311, should be fixed with 4.10.5, please upgrade :+1:
